controller/resource/metricsresource/crud_resource_ops_wrapper.go

```diff
@@ -12,47 +12,40 @@ import (
 type crudResourceOpsWrapperConfig struct {
 	Ops controller.CRUDResourceOps
 
-	ServiceName  string
 	ResourceName string
 }
 
 type crudResourceWrapperOps struct {
 	underlying controller.CRUDResourceOps
 
-	serviceName  string
 	resourceName string
 }
 
 func newCRUDResourceWrapperOps(config crudResourceOpsWrapperConfig) (*crudResourceWrapperOps, error) {
 	if config.Ops == nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.Ops must not be empty")
+		return nil, microerror.Maskf(invalidConfigError, "%T.Ops must not be empty", config)
 	}
 
-	if config.ServiceName == "" {
-		return nil, microerror.Maskf(invalidConfigError, "config.ServiceName must not be empty")
-	}
 	if config.ResourceName == "" {
-		return nil, microerror.Maskf(invalidConfigError, "config.ResourceName must not be empty")
+		return nil, microerror.Maskf(invalidConfigError, "%T.ResourceName must not be empty", config)
 	}
 
 	o := &crudResourceWrapperOps{
 		underlying: config.Ops,
 
-		serviceName:  toCamelCase(config.ServiceName),
```

Comment:
The service name was configured like `aws-operator` and we transformed it to `awsOperator`. I don't remember why we did this, but I figured the metrics have the `app` label anyway, which is `aws-operator`.  So this should just be fine. We drop the service label here and reduce the cardinality.